### PR TITLE
Add helper that creates a grid of curves / surfaces

### PR DESCRIPTION
**Pull request by alexfikl**
Mostly to recreate some configurations from Greengard / Martinsson papers.

### CODE REVIEW

**Comment by alexfikl** [comment]
The `make_mesh_grid` thing should be able to make things like this
Screenshot_20221012_200430

**Comment by alexfikl** [comment]
The dumbbell curve should make things like the classic (also from https://doi.org/10.1016/j.jcp.2004.10.033). Any idea how to add the ripples?

Screenshot_20221012_200629

And the fidget spinner thing is from https://doi.org/10.1137/120866683

Screenshot_20221012_200654

File: meshmode/mesh/generation.py [hunk 1]
```diff
@@ -240,14 +255,16 @@ class NArmedStarfish(WobblyCircle):
...
+    def __init__(self, n_arms: int, amplitude: float, phase: float = 0.0) -> None:
         coeffs = np.zeros(n_arms)
         coeffs[-1] = amplitude
-        super().__init__(coeffs)
+        super().__init__(coeffs, phase=phase)
 
 
 starfish = NArmedStarfish(5, 0.25)
 
+fidget_spinner = NArmedStarfish(3, 1 / 2, phase=np.pi / 2)
```
Comment (alexfikl) [inline meshmode/mesh/generation.py:266]: Name? :cry:

Comment (inducer) [reply meshmode/mesh/generation.py:266]: Fidget starfish?

Comment (alexfikl) [reply meshmode/mesh/generation.py]: Called this a `starfish3` and the old one a `starfish5`. What do you think?

**Comment by inducer** [comment]
> Any idea how to add the ripples?

Looks like they're just y perturbations modulated by something like $(1-x**2)$?

**Comment by alexfikl** [comment]
> Looks like they're just y perturbations modulated by something like (1−x2)?

Yeah, that was it! Thanks! It looks this (with some careful choice of parameters)
<p align="center">

</p>

**Comment by inducer** [comment]
Thx!

**Pull request merged**